### PR TITLE
Add new Container component in Canon

### PR DESCRIPTION
**Pull request by cdedreuille**
## Hey, I just made a Pull Request!

CleanShot 2024-12-06 at 17 54 38@2x

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: packages/canon/src/components/Container/Docs.mdx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+<Title type="h1">Container</Title>
+
+<Text>
+  The container component let you use our default max-width and center the
+  content on the page.
+</Text>
+
+<Title type="h2">Installation</Title>
+<Source code={`yarn install @backstage/canon/container`} language="bash" dark />
```
Comment (Rugvip) [inline packages/canon/src/components/Container/Docs.mdx:20]: Hold up, this doesn't actually work 😅 

Needs to be an import or install of `@backstage/canon`

File: packages/canon/src/components/Container/Docs.mdx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  dark
+/>
+
+<Title type="h3">Responsive padding & margin</Title>
+<Text>
+  The Container component also supports responsive values, making it easy to
+  create responsive designs.
+</Text>
+<Source
+  code={`<Container paddingY={{ xs: 'sm', md: 'md' }}>
```
Comment (Rugvip) [inline packages/canon/src/components/Container/Docs.mdx:65]: Would this be "slightly wider content on xs", or something else? that's what I'd expect anyway.

File: packages/canon/src/components/Grid/Docs.mdx [hunk 3]
```diff
@@ -22,11 +22,11 @@ import { spacingProperties } from '../../layout/sprinkles.css';
   The grid component is made of two parts: the grid container and the grid item. Import all parts and piece them together.
 </Text>
 <Source
-  code={`import * as Grid from "@backstage/canon/grid";
+  code={`import { Grid, GridItem } from "@backstage/canon/grid";
```
Comment (Rugvip) [inline packages/canon/src/components/Grid/Docs.mdx:25]: How come? I do like `Grid.Item`. If you wanna keep that you can use `export Object.assign(Grid, { Item: GridItem })`

**Review by Rugvip — APPROVED** [review]
🚢  & 🇮🇹

File: packages/canon/src/components/Container/Docs.mdx (hunk 1 above)
Comment (cdedreuille) [reply packages/canon/src/components/Container/Docs.mdx:20]: Oh you're right. Radix is actually creating dedicated packages for each of their component. Not sure what I was thinking here :)

File: packages/canon/src/components/Grid/Docs.mdx (hunk 3 above)
Comment (cdedreuille) [reply packages/canon/src/components/Grid/Docs.mdx:25]: Ah you're right. I'll change that.

File: packages/canon/src/components/Container/Docs.mdx (hunk 2 above)
Comment (cdedreuille) [reply packages/canon/src/components/Container/Docs.mdx:65]: This is not the content I'm changing here but the padding on top and bottom. This component essentially set a max-width and center the content. You can only change the width of the content at a top level in the CSS variable.

File: packages/canon/src/components/Container/Docs.mdx (hunk 1 above)
Comment (cdedreuille) [reply packages/canon/src/components/Container/Docs.mdx]: I fixed that.

**Pull request merged**